examples/autoqasm/3_2_magic_state_distillation.ipynb

```diff
@@ -0,0 +1,447 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "6a4927a4-e1ed-4aea-9734-1993f5e741cd",
+   "metadata": {},
+   "source": [
+    "# Magic state distillation and gate teleportation\n",
+    "In fault tolerant quantum computing, Clifford gates have many desirable properties in stabilizer codes. For example, they can be implemented transversally and their noise is contained locally [1]. Therefore, there is a strong preference to include Clifford gates in a logical gateset. However, Clifford gates alone do not form a universal gateset. At least one non-Clifford gate is needed to perform universal quantum computing. Magic states are certain quantum states that enable universal fault-tolerant quantum computing with Clifford gates and preserve the desirable properties of the Clifford gates. With magic states, we can effectively apply non-Clifford gates using only Clifford gates, forming a universal gateset. \n",
+    "\n",
+    "In this notebook, we first demonstration, if we have access to magic states, how to implement a non-Clifford gate with only Clifford gates. This procedure is known as \"gate teleportation\". Because the preparation of magic state in this first demonstration is not fault-tolerant (i.e., it is not prepared under an error correction code), it is subjected to noise. In the second demonstration, we introduce a procedure to create logical magic states on the 5-qubit error correction code [2], called magic state distillation [1]. This procedure is based on post-selection, so the preparation of magic states has a finite probability of failing under this protocol. So, finally, we introduce a \"repeat until success\" protocol [3] that gaurantees every execution prepares magic state successfully. For all the protocols in this notebook, we use AutoQASM to program these procedures, demonstrating AutoQASM's ability to express the classical control flow needed for magic state distillation, showing how AutoQASM enables experimentations on fault-tolerant quantum comuting."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3ba7d657-f5d9-4933-a91b-a870d565d9d4",
+   "metadata": {},
+   "source": [
+    "Let's first import the modules used in this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "ccdb0127-3c13-42b1-980c-c70f2e16fc00",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# general imports\n",
+    "from typing import Dict, List\n",
+    "from collections import Counter\n",
+    "import numpy as np\n",
+    "from collections import defaultdict\n",
+    "\n",
+    "# AWS imports: Import Braket SDK modules\n",
+    "from braket.devices.local_simulator import LocalSimulator\n",
+    "import braket.experimental.autoqasm as aq\n",
+    "import braket.experimental.autoqasm.instructions as ins"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6f902b86-4201-4d0c-a371-6482743f67be",
+   "metadata": {},
+   "source": [
+    "## Gate teleportation\n",
```

Comment:
Nit: 
```suggestion
    "## Noisy magic state preparation: Gate teleportation\n",
```
I see now that I mixed this up with "state teleporation", maybe others like me will find it beneficial to see the correspondence again in the header

Reply:
I changed it to "Apply non-Clifford gates with magic states", which is a more literal title. I also commented (slightly) on the relation between gate teleportation and state teleportation. They are indeed closely related.